### DISCUSSION TITLE
Long read support in seqr

### DISCUSSION THREAD

**Discussion by anjalijain22** [broadinstitute/seqr#3635] [Q&A] [answered]
Hello!

Hope you're well!

Does the seqr development team have any plans of supporting long read analysis in seqr anytime soon?

Thanks!

**Comment by hanars** [comment, marked as answer]
Hi Anjali

We are hoping to add some level of long read support in 2024, but its not yet clear  what the details of that will be

**Reply by anjalijain22** [reply]
Got it, thanks for sharing this!

**Comment by ghost** [comment]
Hi @hanars, et. al.

Last night we processed some single ended long read data from pacbio and loaded it into our dev instance of seqr for testing.  The variant calling was done with deepvariant.  We noticed that the hail pipeline processed it successfully, it works with IGV, and the variant search seems to work.  While we've only had this data available for a few hours to review, I was wondering if you could briefly explain the reasons seqr doesn't support long read analysis.  From our point of view it seems like seqr does support long read data to some degree, but maybe there's some caveats to how the data is presented or filter, etc..

I'm mostly curious what's lacking in the current version that makes the variant search inaccurate for long read data and for analysis by our clinicians and researchers.  What are the drawbacks if we were to load this data into the latest version of seqr?  Is the lack of support partially related to the hail pipeline and the data model?  Will there need to be a separate pipeline for processing this type of data? 

Thanks,
Robert Gilmore

**Reply by hanars** [reply]
Hi Robert,

Its great to hear that you were able to get PacBio to load successfully in seqr! 
The reason seqr does not support long read data is because we have never tried it and with our small engineering team we do not have the resources to help other groups try it out, tweak it and get it working. It feels disingenuous for us to say "seqr supports long read data"  if we are not actually going to ensure that it does and then commit that all future feature changes will not break it. So basically the lack of official support is more of "we aren't sure it works and can't promise it will continue to work" than "this actually can't work". 

That said, if you have gotten it working in seqr I see no reason why it would not be accurate and safe to use! The one concern I would have is that some filtration and annotation might have different meaning in long read than short read data - for example, if a variant is fully absent from gnomad does that means its incredibly rare, or that its a common variant that is not capturable using short read technology? In some preliminary testing we did with long read data internally, we found a problem we kept running into was getting back a lot of noisy variants that were essentially a unannotated due to their absence from many of the tools and databases we were using.  I would also suggest you confirm what the expected distribution of GQ is in your data and possibly adjust how you set quality filters accordingly. However, these are all adaptations you would make to user behavior, not to the application itself. Just make sure your users are aware of these potential drawbacks when assessing variants. 

Regarding the need for a separate pipeline, if you can get your data running through the regular pipeline there is no harm in using it. When our team adds the full support for long read we may decide to break out long read into its own pipeline or not, but have not made that decision. If we did it would likely be because a) we wanted to run some different up-front validation on the data and/or b) we determine that certain reference annotations are so inaccurate that they are more misleading than helpful for long read data and we want to leave them out. However, these would be very minor difference from the main pipeline. 

In short, if you have this data in seqr I encourage you to use it but also encourage you to be extra critical during your analysis.  We would love you hear back from you as your team starts exploring this data if you have any learnings that you think would be helpful to us and others as we go about adding more formal support for this data in seqr.  
Best of luck!

**Reply by ghost** [reply]
Thanks for the thoughtful reply, Hana!  I've shared your response with the GRIS team and we will be sure to get back with you as (or if) we move forward with long read sequences.

**Reply by jxchong** [reply]
@gilmorera fwiw, our group has loaded loaded ONT data into seqr and used it for analysis. We did not do extensive testing but as expected, had no issues with filtration. The main issue is that it's essentially unanalyzable without controls unless you're just looking at specific known variants (e.g. in ClinVar) and/or specific gene due to long-read-specific artifacts and true variants.

**Comment by YasirKusay** [comment]
Hi, I just wanted to ask if there have been any developments to this. We are keen on analysing some of our long read data with our local installation of seqr.

**Reply by hanars** [reply]
Hello, there has been no update on this, but per my [response above](https://github.com/broadinstitute/seqr/discussions/3635#discussioncomment-8467429) if you are running your own seqr instance there is no reason *not* to try and load the long read data that you have. I would also recommend reading [Jessica's reply](https://github.com/broadinstitute/seqr/discussions/3635#discussioncomment-8470015) regarding limitations of that analysis. Unfortunately we have not had the engineering bandwidth to take on full support for long read work in seqr this year, although we do hope to add full support next year pending adequate staffing/ funding

**Reply by YasirKusay** [reply]
Thanks, would it be possible to manually add an LR database that can be used to query our LR data, e.g. `1KGP-LRSC`

**Reply by hanars** [reply]
As long read data is not fully supported in seqr we are unable to provide technical assistance to your team with this endeavour. I encourage you to reach out to the technical team running your seqr instance with any such requests
